engine/updategraph/src/main/java/io/deephaven/engine/updategraph/UpdateGraphProcessor.java

```diff
@@ -1485,19 +1501,42 @@ private void refreshTablesAndFlushNotifications() {
             if (watchdogJob != null) {
                 sched.cancelJob(watchdogJob);
             }
-            final long cycleTime = System.nanoTime() - startTimeNanos;
-            if (cycleTime >= minimumCycleDurationToLogNanos) {
+            jvmIntrospectionContext.endSample();
+            final long cycleTimeNanos = System.nanoTime() - startTimeNanos;
+            if (cycleTimeNanos >= minimumCycleDurationToLogNanos) {
                 if (suppressedCycles > 0) {
                     logSuppressedCycles();
                 }
-                log.info().append("Update Graph Processor cycleTime=").appendDouble(cycleTime / 1_000_000.0)
-                        .append("ms, lockWaitTime=").appendDouble(currentCycleLockWaitTotalNanos / 1_000_000.0)
-                        .append("ms, yieldTime=").appendDouble(currentCycleYieldTotalNanos / 1_000_000.0)
-                        .append("ms, sleepTime=").appendDouble(currentCycleSleepTotalNanos / 1_000_000.0)
-                        .append("ms").endl();
-            } else if (cycleTime > 0) {
-                suppressedCycles++;
-                suppressedCyclesTotalNanos += cycleTime;
+                final long safePointPauseTimeMillis = jvmIntrospectionContext.deltaSafePointPausesTimeMillis();
```

Comment:
We might have reached the point where this logging code adds enough bulk that it should be pulled into a new method, as `logSuppressedCycles` was long ago.